### ISSUE TITLE
run_cmd is always executed - ternary operator is not correctly supported

### ISSUE THREAD

**Issue by ahmad-hamade** [gruntwork-io/terragrunt#1448]
I need to run a conditional `run_cmd` based on the below:

```hcl
locals {
  condition = true
  value     = local.condition ? run_cmd("echo", "A") : run_cmd("echo", "B")
}
```

Running `terragrunt plan/apply` will execute always both commands:

```
[terragrunt] 2020/11/29 14:45:23 Running command: echo A
A
[terragrunt] 2020/11/29 14:45:23 run_cmd output: [A
]
[terragrunt] 2020/11/29 14:45:23 Running command: echo B
B
[terragrunt] 2020/11/29 14:45:23 run_cmd output: [B
]
```

The expected behavior is to run only `run_cmd("echo", "A")`

I'm using terragrunt version v0.26.7

**Comment by brikis98** [comment]
Ah, yea, that's a bug. Probably related to #1427: something about our parsing is probably executing `run_cmd`, preventing the short circuiting behavior. If anyone has a chance to dig in and fix the issue, a PR is very welcome, as we're buried right now.

**Comment by brikis98** [comment]
Since this is most likely the same issue as #1427, closing in favor of that issue.

**Comment by MeNsaaH** [comment]
@brikis98  I don't think this is the same issue as the one you closed in favor of. The issue here is that when used with a conditional operator, only one of the runcmd commands should be executed. But both are getting executed.

I'm currently on `0.35.10` and the same issue happens.

**Comment by Tailzip** [comment]
This is happening for me as well, on `0.35.13`

**Comment by geekofalltrades** [comment]
Also happening for me on v0.36.0. This was not fixed by the fix for the other issue (https://github.com/gruntwork-io/terragrunt/issues/1427#issuecomment-903719132).

**Comment by tkent** [comment]
A quick chime in. This is definitely still happening as of `v.0.37.1`. If you are really trying to to avoid  executing some command based on environment factors, the work around is pretty ugly. For example:

```terraform
specified_public_ip = get_env("PUBLIC_IP")
cmd_args            = (local.specified_public_ip != null
    ? ["printf", local.specified_public_ip]
    : ["curl", "ifconfig.me/ip"]
  )

  effective_public_ip = run_cmd(local.cmd_args...)
```

**Comment by DmytroRomantsovM** [comment]
It's still reproducible in Terragrunt 0.82.4. Very non-intuitive behaviour. Shall the issue be reopened?

**Comment by wakeful** [comment]
Reopening this issue - **terragrunt** currently does not properly support the ternary operator within the locals block.
As this isn't a high-priority item, I've labeled it as contributions-requested in case a community member is interested in taking it on.

**Comment by cmeury** [comment]
I believe this is still relevant.

**Comment by cmeury** [comment]
Still relevant.

**Comment by cmeury** [comment]
Still relevant, there's even a linked PR.

**Comment by yakyouk** [comment]
FWIW here's a workaround:

```
myvar = local.should_evaluate_myvar ? (
  [for k in (local.should_evaluate_myvar?[0]:[]):(  # prevent greedy evaluation
    run_cmd("pwsh","-NoProfile","-c","'myvalue'")
  )][0]
) : null
```
